src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceAction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+
+/**
+ * GeoIp datasource creation action
+ */
+public class PutDatasourceAction extends ActionType<AcknowledgedResponse> {
+    /**
+     * Put datasource action instance
+     */
+    public static final PutDatasourceAction INSTANCE = new PutDatasourceAction();
+    /**
+     * Name of a put datasource action
+     */
+    public static final String NAME = "cluster:admin/geospatial/datasource";
```

Comment:
why cluster admin?

Reply:
This is mostly used for permission control as long as I know. I thought it would be better to give ip2geo datasource related permission to admin because, 1. ip2geo datasource creation is resource expensive, 2. there are not many different ip2geo datasource for each use case. It will be mostly one or two ip2geo datasource which will be shared by many ip2geo processors.